misk/src/main/kotlin/misk/web/actions/LivenessCheckAction.kt

```diff
@@ -17,11 +21,17 @@ class LivenessCheckAction @Inject internal constructor(
   @Get("/_liveness")
   @ResponseContentType(MediaTypes.TEXT_PLAIN_UTF8)
   fun livenessCheck(): Response<String> {
-    val isAlive = services.all {
-      it.state() != State.FAILED && it.state() != State.TERMINATED
+    val failedServices = services.filter {
+      failedServiceStates.contains(it.state())
+    }
+
+    if (failedServices.isEmpty()) {
+      return Response("", statusCode = 200)
     }
 
-    // TODO(jgulbronson) - Should return an empty body
-    return Response("", statusCode = if (isAlive) 200 else 503)
+    for (service in failedServices) {
+      logger.error("Service failed: $service")
```

Comment:
this is just the service , not the reason?

Reply:
For some services, it's the service and state, for example `AbstractExecutionThreadService`:

```java
 public String toString() {
    return serviceName() + " [" + state() + "]";
  }
```

This isn't guaranteed by the `Service` interface, though.  Also, `serviceName()` isn't part of the interface either, so `toString()` is the only option available to get the name.

For a more detailed reason we could use `Service.failureCause`...but that throws if the service state isn't `FAILED`, so I'd need to write a helper method.